### PR TITLE
Slash commands: Group several similar commands into single commands

### LINKED ISSUE #512: Slash commands: Group several similar commands into single commands
Complete the following REQUIRED checkboxes:
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)

-   [x] The title of this issue follows the `command name: brief description of request` format, e.g. `/help: add optional @user parameter`

The following checkbox is OPTIONAL:

-   [x] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**

Several similar commands can be grouped into single commands. For instance, we currently have the following commands to search in a specific site:
- `/google` to search in google
- `/searchmdn` to search in MDN docs
- `/searchtop` to search in TOP

~~I suggest we group those commands into a single command `search` with the `site` as an option. The `site` option would contain a dropdown containing the site the user wants to search in.~~

I suggest we group those commands into a single command, with each site as a subcommand.

**2. Acceptance Criteria:**

- [ ] All slash commands that can be grouped are grouped into single commands

**3. Additional Information:**

### PR DESCRIPTION
**Pull request by cakegod**
## Because

We can group multiple similar commands into single commands, making commands easier to remember.

## This PR

- Groups several similar commands into single commands

## Issue

Closes #512

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Callbacks command: Update verbiage`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If this PR adds new features or functionality, I have added new tests
-   [x] If applicable, I have ensured all tests related to any command files included in this PR pass, and/or all snapshots are up to date

### CODE REVIEW

**Comment by cakegod** [comment]
@01zulfi The grouping of the search commands is done. I'm unsure if we want to group more commands.

File: new-era-commands/slash/search.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    .setDescription(site.description)
+    .addStringOption((option) =>
+      option
+        .setName("prompt")
+        .setDescription("term to search")
+        .setRequired(true),
+    )
+    .addUserOption((option) =>
+      option.setName("user").setDescription("user to ping"),
+    );
```
Comment (MaoShizhong) [inline new-era-commands/slash/search.js:38]:
```suggestion
const searchSlashCommand = new SlashCommandBuilder()
  .setName('search')
  .setDescription('Search in a specific site');

Object.values(searchSites).forEach((site) => {
  searchSlashCommand.addSubcommand((subcommand) =>
    subcommand
      .setName(site.name.toLowerCase())
      .setDescription(site.description)
      .addStringOption((option) =>
        option
          .setName('prompt')
          .setDescription('term to search')
          .setRequired(true),
      )
      .addUserOption((option) =>
        option.setName('user').setDescription('user to ping'),
      ),
  );
});
```
Factoring out the SlashCommandBuilder instantiation allows you to auto-apply any new search commands by only adding to the searchSites object, without needing to also remember to append `.addSubcommand(createCommand(searchSites[nameOfSite]))`.

File: new-era-commands/slash/search.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName("search")
+    .setDescription("Search in a specific site")
+    .addSubcommand(createCommand(searchSites.google))
+    .addSubcommand(createCommand(searchSites.mdn))
+    .addSubcommand(createCommand(searchSites.top)),
```
Comment (MaoShizhong) [inline new-era-commands/slash/search.js:46]:
```suggestion
  data: searchSlashCommand,
```
Which would then mean the exported object only needs to contain this as the `.data` value.

If people *really* want to have all the logic housed directly within the exported object, then an IIFE can be used I guess.

**Review by MaoShizhong — COMMENTED** [review]
I like the link button interface. Some refactoring suggestions to reduce duplication.

File: new-era-commands/slash/search.js (hunk 1 above)
Comment (cakegod) [reply new-era-commands/slash/search.js:38]: This was initially the idea, but it felt a bit unclear as the structure wasn't similar to the other commands.

Comment (MaoShizhong) [reply new-era-commands/slash/search.js:38]: You could always crack out the ol' IIFE to keep the structure within the file in line with others, where the slash command is defined/modified directly within `.data` property.

Comment (MaoShizhong) [reply new-era-commands/slash/search.js:38]: Could do, although with an array, you'd need to `.find` in the execute callback to get the right object as opposed to just accessing the object key.

Having just said that, the `name` field for each object is redundant given it's just the key without forced lowercase. What I used with `Objects.values` can be changed to keys or entries.

Comment (cakegod) [reply new-era-commands/slash/search.js:38]: Read the edit.

Comment (MaoShizhong) [reply new-era-commands/slash/search.js:38]: Maps are lovely 👌

**Review by MaoShizhong — DISMISSED** [review]

**Comment by MaoShizhong** [comment]
Would like to see what Xari says as well, just to be sure.

**Review by 01zulfi — APPROVED** [review]
lgtm 🚀

**Pull request merged**